### ISSUE TITLE
 Interface 'CookieAgent<BaseAgent>' incorrectly extends interface 'BaseAgent'

### ISSUE THREAD

**Issue by BeataKr** [3846masa/http-cookie-agent#37]
I'm facing with ts problem

```
node_modules/http-cookie-agent/index.d.ts:14:11 - error TS2430: Interface 'CookieAgent<BaseAgent>' incorrectly extends interface 'BaseAgent'.
  'CookieAgent<BaseAgent>' is assignable to the constraint of type 'BaseAgent', but 'BaseAgent' could be instantiated with a different subtype of constraint 'any'.

14 interface CookieAgent<BaseAgent extends http.Agent> extends BaseAgent {
```

My configuration is kind of simple:

```
import axios from 'axios';
import { HttpCookieAgent } from 'http-cookie-agent';
import { CookieJar } from 'tough-cookie';

const jar = new CookieJar();
this.axios.defaults.httpAgent = new HttpCookieAgent({
        jar,
        keepAlive: true,
      });
```

Used version of packs:
```
"dependencies": {
    "axios": "^0.24.0",   
    "http-cookie-agent": "^1.0.3",   
    "tough-cookie": "^4.0.0",
    },
"devDependencies": {
    "@types/tough-cookie": "^4.0.1",   
    "typescript": "4.4.4"
  }
```

Any idea how to resolve problem?

**Comment by BeataKr** [comment]
This is not a bug, but a kind of future feature that can be done. An error can be disabled by
```
{
  "compilerOptions": {
    "skipLibCheck": true,
    ...
}
```

**Comment by 3846masa** [comment]
Thanks for reporting.

It has been fixed in [v1.0.4](https://github.com/3846masa/http-cookie-agent/releases/tag/v1.0.4).
